beps/0003-auth-architecture-evolution/README.md

```diff
@@ -201,19 +211,33 @@ export default createBackendPlugin({
 
 This is expected to be the pattern for the vast majority of plugins.
 
-#### A plugin with an endpoint that only allows cookie auth
+#### A plugin with a cookie-based authentication endpoint
 
 ```ts
 export default createBackendPlugin({
   pluginId: 'techdocs',
   register(env) {
     env.registerInit({
       deps: {
+        auth: coreServices.auth,
+        httpAuth: coreServices.httpAuth,
         http: coreServices.httpRouter,
       },
-      async init({ http }) {
+      async init({ auth, httpAuth, http }) {
+        const router = Router();
+
+        // Endpoint that sets the cookie for the user
+        router.get('/cookie', async (req, res) => {
+          await httpAuth.issueUserCookie(req);
```

Comment:
hm the method signature defined below for this is 
```tsx
issueUserCookie(
    res: Response,
    options?: {
      // If credentials are not provided, they will be read from the request
      credentials?: BackstageCredentials<BackstageUserPrincipal>;
    },
  ): Promise<void>;
```

so it looks like this should take `res` as a parameter but also `req` to extract default credentials according to the comment within `options`?

Reply:
`res.req` to the rescue 😁

Reply:
ah TIL - then this should be `await httpAuth.issueUserCookie(res);` instead?

Reply:
There are some situations where we want to issue credentials that don't arrive in the auth header, but that would also be valid

Reply:
ah sorry - I was a bit confused here on how `httpAuth.issueUserCookie(req)` sets the auth cookie on the resp and didn't realize `req` and `res` both have a reference to each other 😅

Reply:
hm but now that I think about it, won't `httpAuth.issueUserCookie(req)` be a type error since the first param expects a `res: Response`

Reply:
Ah yep the example here is wrong, should be `res` :+1: